tests/end_to_end/end_to_end.cpp

```diff
@@ -4192,4 +4207,56 @@ test_sample_perf_buffer_test(ebpf_execution_type_t execution_type)
 
 DECLARE_ALL_TEST_CASES("test-sample-perfbuffer", "[end_to_end]", test_sample_perf_buffer_test);
 DECLARE_ALL_TEST_CASES("bindmonitor-perfbuffer", "[end_to_end]", bindmonitor_perf_buffer_test);
-DECLARE_ALL_TEST_CASES("negative_perf_buffer_test", "[end_to_end]", negative_perf_buffer_test);
\ No newline at end of file
+DECLARE_ALL_TEST_CASES("negative_perf_buffer_test", "[end_to_end]", negative_perf_buffer_test);
+
+TEST_CASE("signature_checking", "[end_to_end]")
+{
+    _test_helper_end_to_end test_helper;
+    _ebpf_platform_code_integrity_test_signing_enabled = false;
+    test_helper.initialize();
+    _ebpf_platform_code_integrity_test_signing_enabled = true;
+
+    const char* eku_list[] = {
+        "1.3.6.1.5.5.7.3.3",      // Code signing
```

Comment:
This string is hard coded several times in this PR.  Can it be in a define in a header file?